for_curators/sample_local_dataset_upload_dev.py.ipynb

```diff
@@ -0,0 +1,305 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 120,
+   "id": "7c8c55fa",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "api_key_file = \"path/to/api-key.txt\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "3e84d52e",
+   "metadata": {},
+   "source": [
+    "<font color='#bc00b0'>(Required) Provide the absolute path to the h5ad datafile to upload</font>"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 121,
+   "id": "f46e1874",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "filename = \"/absolute/path/to-datafile.h5ad\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "104aea94",
+   "metadata": {},
+   "source": [
+    "<font color='#bc00b0'>(Required) Enter your chosen `curator_tag`, which will serve as a unique identifier _within this Collection_ for the resultant Dataset. **Must possess the '.h5ad' suffix**.</font>\n",
+    "    \n",
+    "_We recommmend using a tagging scheme that 1) makes sense to you, and 2) will help organize and facilitate your \n",
+    "automation of future uploads for adding new Datasets and replacing existing Datasets._"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 122,
+   "id": "b2198be1",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "curator_tag = \"arbitrary/tag/choosen-by-you.h5ad\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "463870f1",
+   "metadata": {},
+   "source": [
+    "<font color='#bc00b0'>(Required) Enter the id of the Collection to which you wish to add this datafile as a Dataset</font>\n",
+    "\n",
+    "_You can only add/replace Datasets in private Collections or revision Collections._"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 123,
+   "id": "f8bb640a",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "collection_id = \"01234567-89ab-cdef-0123-456789abcdef\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "352deec2",
+   "metadata": {},
+   "source": [
+    "### Import dependencies"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "17d70d36",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import boto3\n",
+    "import os\n",
+    "import requests\n",
+    "import threading\n",
+    "from botocore.credentials import RefreshableCredentials\n",
+    "from botocore.session import get_session\n",
+    "from datetime import datetime, timezone"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "1708156b",
+   "metadata": {},
+   "source": [
+    "### Use API key to obtain a temporary access token"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 125,
+   "id": "7f3abf8b",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "api_key = open(api_key_file).read().strip()  \n",
+    "access_token_headers = {\"x-api-key\": api_key}\n",
+    "access_token_url = \"https://api.cellxgene.dev.single-cell.czi.technology/curation/v1/auth/token\"\n",
```

Comment:
Do the curators have access to dev?

Reply:
Jason does. This will be modified for the final version in prod.